python/mlx/optimizers.py

```diff
@@ -498,3 +498,205 @@ def apply_single(
         if weight_decay > 0:
             parameter = (1 - lr * weight_decay) * parameter
         return parameter - lr * mx.sign(c)
+
+
+class LRScheduler:
+    r"""Base class for learning rate schedulers.
+
+    Args:
+        optimizer (Optimizer): The optimizer for which to adjust the learning rate.
+        last_epoch (int, optional): The index of the last epoch. Default: -1.
+    """
+
+    def __init__(self, optimizer: Optimizer, last_epoch: int = -1):
+        if not isinstance(optimizer, Optimizer):
+            raise TypeError(f"{type(optimizer).__name__} is not an Optimizer")
+        self.optimizer = optimizer
+        self.last_epoch = last_epoch
+        self.base_lr = optimizer.learning_rate
+        self.step(last_epoch)
+
+    def get_lr(self) -> float:
+        raise NotImplementedError
+
+    def step(self, epoch: int = None):
+        if epoch is None:
+            epoch = self.last_epoch + 1
+        self.last_epoch = epoch
+        self.optimizer.learning_rate = self.get_lr()
+
+
+class StepLR(LRScheduler):
+    r"""Decays the learning rate by a factor of gamma every step_size epochs.
+
+    Args:
+        optimizer (Optimizer): The optimizer for which to adjust the learning rate.
+        step_size (int): Period of learning rate decay.
+        gamma (float, optional): Multiplicative factor of learning rate decay. Default: 0.1.
+        last_epoch (int, optional): The index of the last epoch. Default: -1.
+    """
+
+    def __init__(
+        self,
+        optimizer: Optimizer,
+        step_size: int,
+        gamma: float = 0.1,
+        last_epoch: int = -1,
+    ):
+        self.step_size = step_size
+        self.gamma = gamma
+        super().__init__(optimizer, last_epoch)
```

Comment:
Usually, we put the call of parent's constructor at the top of the descendants.

Reply:
Fixed, thanks!